sw/host/opentitantool/src/command/image.rs

```diff
@@ -119,7 +127,23 @@ impl CommandDispatch for ManifestUpdateCommand {
             image.overwrite_manifest(def)?;
         }
 
-        // TODO: Add signature update.
+        if self.signature_file.is_some() != self.key_file.is_some() {
+            bail!("Signature cannot be updated without both a signature file and key file");
+        }
```

Comment:
You're right, the signed region starts immediately after the signature and extends until the end of the image and updating the modulus and the signature would invalidate the signature (assuming modulus is different than what the manifest has).

But, this might be useful for resigning a manifest with a different key after a minor modification (please see my comment above).

Reply:
I still don't understand why I need both the signature file and the key file at the same time.

Prior to signing, I need to update the manifest to contain a bunch of basic info _and_ the modulus.  I won't have a signature file at that time.

After signing, when I _do_ have the signature file, I've already injected the modulus (prior to signing) and changing it will invalidate the signature.  Updating it to the same value is wasted effort.

Reply:
> After signing, when I _do_ have the signature file, I've already injected the modulus (prior to signing) and changing it will invalidate the signature. Updating it to the same value is wasted effort.

You're right it doesn't make sense for the actual signing flow. I was thinking about resigning a manifest with multiple keys in a dev/test workflow. It's such a contrived case, please ignore my comment :)